### PR TITLE
chore: set circle-ci cache key basing on pnpm lock file

### PR DESCRIPTION
**Pull request by UncleBill**
resolves https://app.circleci.com/pipelines/github/DimensionDev/Maskbook/10714/workflows/40b90427-73aa-4ac3-952b-20c8e4f9ac6f/jobs/26315

cc @zhouhanseng

### CODE REVIEW

**Comment by nuanyang233** [comment]
https://app.circleci.com/pipelines/github/DimensionDev/Maskbook/10716/workflows/6e73104d-dcfd-426d-aaac-6717c604b35a/jobs/26318
It doesn't look like it's been solved

**Comment by UncleBill** [comment]
> [app.circleci.com/pipelines/github/DimensionDev/Maskbook/10716/workflows/6e73104d-dcfd-426d-aaac-6717c604b35a/jobs/26318](https://app.circleci.com/pipelines/github/DimensionDev/Maskbook/10716/workflows/6e73104d-dcfd-426d-aaac-6717c604b35a/jobs/26318)
> It doesn't look like it's been solved

Indeed. Caching might be not the reason.

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,13 +22,13 @@ jobs:
       - restore_cache:
           name: Restore PNPM Package Cache
           keys:
-            - v3-maskbook-cache-pnpm
+            - v3-maskbook-cache-pnpm-{{ checksum "pnpm-lock.yaml" }}
       - run:
           name: Install Dependencies
           command: npx pnpm install --frozen-lockfile --store-dir ~/.pnpm/
       - save_cache:
           name: Save NPM Package Cache
```
Comment (septs) [inline .circleci/config.yml:30]:
```suggestion
          name: Save PNPM Package Cache
```

**Review by septs — CHANGES_REQUESTED** [review]

**Review by zhouhanseng — APPROVED** [review]

**Review by zhouhanseng — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,13 +22,13 @@ jobs:
       - restore_cache:
           name: Restore PNPM Package Cache
           keys:
-            - v3-maskbook-cache-pnpm
+            - v3-maskbook-cache-pnpm-{{ checksum "pnpm-lock.yaml" }}
```
Comment (Jack-Works) [inline .circleci/config.yml:25]: pnpm has a "global" store and a "local store". You can think of "local store" as a hard link to the "global store". And here we're catching the global store (cache) not the local store (node_modules) so I think we should not add a checksum to the package-lock.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (UncleBill) [reply .circleci/config.yml:25]: I see.

**Comment by UncleBill** [comment]
node_modules are fresh after every dependencies installation, no need to set cache key basing on lock file.

**Pull request closed without merging**